public/components/security/roles-mapping/roles-mapping.tsx

```diff
@@ -18,8 +18,14 @@ import { Rule } from '../rules/types/rule.type';
 import { Role } from '../roles/types/role.type';
 import RolesServices from '../roles/services';
 import RulesServices from '../rules/services';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
 
-export const RolesMapping = () => {
+const mapStateToProps = state => ({ currentPlatform: state.appStateReducers.currentPlatform });
```

Comment:
To keep the component as a function, you could replace the HOC with the `useSelector` hook from 'react-redux'

```
import { useSelector, useDispatch } from 'react-redux';

export const RolesMapping  = () => {
  // Extract the variable form Redux store
  const currentPlatform= useSelector(state => state.appStateReducers.currentPlatform);
// ... code
}